rc.js

```diff
@@ -29,6 +29,8 @@ if (env.npm_config_argv) {
 // Get the configuration
 module.exports = function (pkg) {
   var pkgConf = pkg.config || {}
+  var source_build = env.npm_config_build_from_source;
+  var buildFromSource = source_build === pkg.name || (source_build === true || source_build === 'true');
```

Comment:
Can you explain these cases?

Reply:
- `pkg.name` is used by some people to only build from source a specific module
- `"true"` is the usual scenario (this is what I need for my builds)
- `true` is in case npm decides to cast the type to Boolean

They are in use inside node-pre-gyp https://github.com/mapbox/node-pre-gyp/blob/67c156ee45804f1aeca2dfed0f3939ebe04b89ad/lib/install.js#L198

Reply:
I don't see how it can ever be a boolean, seeing as the value comes from `process.env`.

Reply:
Here is a reference for somebody using an explicit package name instead of a Boolean
https://git.cse.msu.edu/schne434/step11/blob/1c4d1c87558b523d42d82abc0478acc50e050354/node_modules/fsevents/node_modules/node-pre-gyp/README.md#L83